tests/pos/i9530.scala

```diff
@@ -0,0 +1,37 @@
+trait Food
+case class Banana(color: String) extends Food
+
+trait Diet[A <: Animal]:
+  type F <: Food
+  def food: Seq[F]
+
+trait Animal
+object Animal:
+  extension [A <: Animal](using diet: Diet[A])(animal: A) def food1 = diet.food
+  extension [A <: Animal](animal: A)(using diet: Diet[A]) def food2 = diet.food
+
+extension [A <: Animal](using diet: Diet[A])(animal: A) def food3 = diet.food
+extension [A <: Animal](animal: A)(using diet: Diet[A]) def food4 = diet.food
+
+trait Monkey extends Animal
+
+given Diet[Monkey] with
+  type F = Banana
+  def food: Seq[Banana] = Seq(new Banana("yellow"), Banana("green"))
+
+trait FoodOps
+given FoodOps with
+  extension [A <: Animal](using diet: Diet[A])(animal: A) def food5 = diet.food
+  extension [A <: Animal](animal: A)(using diet: Diet[A]) def food6 = diet.food
+
+
+val monkey = new Monkey {}
+
+val foods = Seq(
+  monkey.food1, // doesn't compile
```

Comment:
Now everything should compile ;)
We should probably also add cases for extensions hidden in implicits which are themselves defined in a companion object